libgalois/src/PropertyGraph.cpp

```diff
@@ -483,16 +586,39 @@ katana::Result<void>
 katana::PropertyGraph::DoWrite(
     tsuba::RDGHandle handle, const std::string& command_line,
     tsuba::RDG::RDGVersioningPolicy versioning_action) {
+  std::unique_ptr<tsuba::FileFrame> topology_res = nullptr;
```

Comment:
The logic in this function needs to improve. First, you shouldn't be declaring variables at the top as null, because that opens up a possibility for mess ups. For example, if one of the if-conditions below is false, you'll be passing a null ptr to rdg_.Store(). can rdg_.Store() work with nullptrs?

One way to rewrite this would be:
```
if (rdg_.A.Valid() || rdg_.B.Valid() || rdg_.C.Valid()) {
    // retrun error; I'm not sure if we should return error in this case.
}
std::unique_ptr<tsuba::FileFrame> topo_res = KATANA_CHECKED(WriteTopology(topology()));
....
```